src/components/organisms/edit-user-modal/index.tsx

```diff
@@ -1,93 +1,101 @@
-  const [email, setEmail] = useState(user.email)
-  const [first_name, setFirstName] = useState(user.first_name)
-  const [last_name, setLastName] = useState(user.last_name)
+  const { mutate, isLoading } = useAdminUpdateUser(user.id)
+  const { register, handleSubmit, reset } = useForm<EditUserModalFormData>()
   const notification = useNotification()
 
-  const submit = () => {
-    setIsLoading(true)
-    Medusa.users
-      .update(user.id, {
-        first_name,
-        last_name,
-      })
-      .then((res) => onSubmit())
-      .catch((err) => notification("Error", getErrorMessage(err), "error"))
+  useEffect(() => {
+    reset(mapUser(user))
+  }, [user])
 
-    handleClose()
+  const onSubmit = (data: EditUserModalFormData) => {
+    mutate(data, {
+      onSuccess: () => {
+        notification("Success", `User was updated`, "success")
+        onSuccess()
+      },
+      onError: (error) => {
+        notification("Error", getErrorMessage(error), "error")
+      },
+      onSettled: () => {
+        handleClose()
+      },
+    })
   }
 
   return (
     <Modal handleClose={handleClose}>
-      <Modal.Body>
-        <Modal.Header handleClose={handleClose}>
-          <span className="inter-xlarge-semibold">Edit User</span>
-        </Modal.Header>
-        <Modal.Content>
-          <div className="w-full flex mb-4">
-            <InputField
-              label="First Name"
-              placeholder="First name..."
-              onChange={(e) => setFirstName(e.target.value)}
-              value={first_name}
-              className="mr-4"
-            />
-            <InputField
-              label="Last Name"
-              placeholder="Last name..."
-              onChange={(e) => setLastName(e.target.value)}
-              value={last_name}
-            />
-          </div>
-          <InputField label="Email" disabled value={email} />
-        </Modal.Content>
-        <Modal.Footer>
-          <div className="w-full flex justify-end">
-            <Button
-              variant="ghost"
-              size="small"
-              onClick={handleClose}
-              className="mr-2"
-            >
-              Cancel
-            </Button>
-            <Button
-              loading={isLoading}
-              variant="primary"
-              size="small"
-              onClick={submit}
-            >
-              Save
-            </Button>
-          </div>
-        </Modal.Footer>
-      </Modal.Body>
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <Modal.Body>
+          <Modal.Header handleClose={handleClose}>
+            <span className="inter-xlarge-semibold">Edit User</span>
+          </Modal.Header>
+          <Modal.Content>
+            <div className="w-full flex gap-x-base mb-base">
+              <InputField
+                label="First Name"
+                placeholder="First name..."
+                {...register("first_name", { required: true })}
```

Comment:
question: In some places there is both required on the input and in the register, in other places it is only set in the register. should we make the input required too?